ykpack/src/types.rs

```diff
@@ -108,21 +108,29 @@ impl Display for DefId {
     }
 }
 
+/// Bits in the `flags` bitfield in `Body`.
+pub mod bodyflags {
+    pub const TRACE_HEAD: u8 = 1;
+    pub const TRACE_TAIL: u8 = 1 << 1;
```

Comment:
`pub const TRACE_TAIL: u8 = 2;`
or
`pub const TRACE_TAIL: u8 = 0x10;`
?

Reply:
I would probably write it like this:
```
pub const TRACE_HEAD: u8 = 0x01;
pub const TRACE_TAIL: u8 = 0x10;
```
So it's immediately clear these are flags, and doesn't require any additional decoding by someone reading the code.

Reply:
I hate to break it to you, but `0x10` is not 2 :P

But more seriously, the `<<` notation is more conventional, because it means you don't have to calculate the power of 2 manually. Admittedly that's not a big issue for a `u8` bitfield, but still.

Reply:
> I hate to break it to you, but 0x10 is not 2 :P
Haha, sorry, I meant `0b02`. :D

> But more seriously, the << notation is more conventional
If this is the convention then that's fair enough. I personally just find it easier to read `0b10` instead of `1 << 1`.

Reply:
`0b` is fine for a `u8`, but now consider a u32:
`0b00000000100000000000000000000000` vs. `1<<23`.

Or a `u64`...

Reply:
Can a trace be both `TRACE_HEAD` and `TRACE_TAIL`? If not, then there's no need to worry about bits in quite this way: just a numeric differentiation would be enough, right?

Reply:
In theory it could happen, but I doubt it. I've used a flags bitfield to allow for additional flags in the future.